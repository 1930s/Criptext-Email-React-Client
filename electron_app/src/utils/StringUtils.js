const { appDomain } = require('./const');
const { HTMLTagsRegex } = require('./RegexUtils');

const removeDomainFromEmail = (email, domain) => {
  return email.replace(`@${domain}`, '');
};

/* To export
   ----------------------------- */
const removeAppDomain = email => {
  return removeDomainFromEmail(email, appDomain);
};

const removeHTMLTags = string => {
  return string.replace(HTMLTagsRegex, '');
};

module.exports = {
  removeAppDomain,
  removeHTMLTags
};
