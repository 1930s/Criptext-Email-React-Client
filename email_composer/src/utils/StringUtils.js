export const pasteSplit = data => {
  const separators = [
    ',',
    ';',
    '\\(',
    '\\)',
    '\\*',
    '/',
    ':',
    '\\?',
    '\n',
    '\r'
  ];
  return data.split(new RegExp(separators.join('|'))).map(d => d.trim());
};

export const getTwoCapitalLetters = string => {
  const strings = string.split(' ');
  if (strings.length === 1) {
    return strings[0].slice(0, 2).toUpperCase();
  }
  return (strings[0].charAt(0) + strings[1].charAt(0)).toUpperCase();
};

export const replaceAllOccurrences = (text, search, replacement) => {
  return text.split(search).join(replacement);
};

export const convertToHumanSize = (bytes, si) => {
  const thresh = si ? 1000 : 1024;
  if (Math.abs(bytes) < thresh) {
    return bytes + ' B';
  }
  const units = si
    ? ['KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
    : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
  let u = -1;
  do {
    bytes /= thresh;
    ++u;
  } while (Math.abs(bytes) >= thresh && u < units.length - 1);
  return bytes.toFixed(1) + ' ' + units[u];
};

export const removeActionsFromSubject = subject => {
  const actions = ['Re:', 'RE:', 'Fw:', 'FW:', 'Fwd:', 'FWD:'];
  return deletePrefixingSubstrings(actions, subject);
};

const deletePrefixingSubstrings = (substrings, subject) => {
  const substringToDelete = hasAnySubstring(substrings, subject);
  if (substringToDelete) {
    subject = deleteSubstring(substringToDelete, subject);
    return deletePrefixingSubstrings(substrings, subject);
  }
  return subject;
};

const hasAnySubstring = (substrings, string) => {
  return substrings.find(substring => {
    return string.indexOf(substring) === 0;
  });
};

const deleteSubstring = (substring, string) => {
  return string.replace(substring, '').trim();
};
