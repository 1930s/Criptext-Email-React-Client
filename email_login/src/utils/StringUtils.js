export const censureEmailAddress = address => {
  const [user, domain] = address.split('@');
  const domainLevels = domain.split('.');
  const lastLevel = domainLevels.pop();
  const character = '*';
  const censoredUser = replaceCharacters(user, [0], character);
  const censoredDomain = `${replaceCharacters(
    domainLevels.join('.'),
    [0, -1],
    character
  )}.${lastLevel}`;
  return `${censoredUser}@${censoredDomain}`;
};

export const replaceCharacters = (string, positionsToExclude, character) => {
  if (typeof positionsToExclude === 'string') {
    return positionsToExclude === 'all'
      ? string
      : positionsToExclude === 'none'
        ? character.repeat(string.length)
        : string;
  }
  if (Array.isArray(positionsToExclude)) {
    const sanitizedPositions = positionsToExclude.map(position => {
      return position < 0 ? string.length + position : position;
    });
    let replacedString = '';
    for (let i = 0; i < string.length; i++) {
      const newCharacter = sanitizedPositions.includes(i)
        ? string.charAt(i)
        : character;
      replacedString += newCharacter;
    }
    return replacedString;
  }
};
