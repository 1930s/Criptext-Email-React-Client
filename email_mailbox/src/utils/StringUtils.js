export const removeActionsFromSubject = subject => {
  const actions = ['Re:', 'RE:'];
  return deleteSubstringsFirstPosition(actions, subject);
};

export const deleteSubstringsFirstPosition = (substrings, subject) => {
  const substringToDelete = hasAnySubstring(substrings, subject);
  if (substringToDelete) {
    subject = deleteSubstring(substringToDelete, subject);
    return deleteSubstringsFirstPosition(substrings, subject);
  }
  return subject;
};

const hasAnySubstring = (substrings, string) => {
  return substrings.find(substring => {
    return string.indexOf(substring) === 0;
  });
};

const deleteSubstring = (substring, string) => {
  return string.replace(substring, '').trim();
};
