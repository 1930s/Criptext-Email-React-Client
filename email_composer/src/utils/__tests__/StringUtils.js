/* eslint-env node, jest */

import * as utils from '../StringUtils.js';

describe('string utils:', () => {
  it('remove criptext domain to criptext email', () => {
    const email = 'erika@criptext.com';
    const state = utils.removeCriptextDomain(email);
    expect(state).toEqual('erika');
  });

  it('remove criptext domain to any email', () => {
    const email = 'erika@signal.com';
    const state = utils.removeCriptextDomain(email);
    expect(state).toEqual(email);
  });

  it('remove HTML tags', () => {
    const text =
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry.';
    const numbers = '1, 2, 4';
    const test = `<p><span>${text}</span><a>${numbers}</a></p>`;
    const state = utils.removeHTMLTags(test);
    expect(state).toEqual(text + numbers);
  });
});
