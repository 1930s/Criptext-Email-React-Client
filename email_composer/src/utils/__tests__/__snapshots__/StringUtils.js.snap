// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string utils: Parse email by separators 1`] = `
Array [
  "julian_adams@signal.com",
  "erika@criptext.com",
  "gianni-carlo@signal.com",
  "daniel_tigse@signal.com",
  "pedro_iniguez@criptext.com",
  "gabriel@signal.com",
  "allison@signal.com",
  "natasha@criptext.com",
  "julian@signal.com",
  "juan_piguave@signal.com",
  "erick@criptext.com",
]
`;
